pkg/kubectl/cmd/util/factory.go

```diff
@@ -354,6 +354,19 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 					return nil, errors.New("provided options object is not a PodLogOptions")
 				}
 				return c.Pods(t.Namespace).GetLogs(t.Name, opts), nil
+
+			case *api.ReplicationController:
+				opts, ok := options.(*api.PodLogOptions)
+				if !ok {
+					return nil, errors.New("provided options object is not a PodLogOptions")
+				}
+				selector := labels.SelectorFromSet(t.Spec.Selector)
+				pod, err := GetFirstPod(c, t.Namespace, selector)
```

Comment:
Printing a message that includes the pod name sounds sufficient. I mainly want to avoid a situation where running
```
kubectl logs rc/foo
...
kubectl logs rc/foo
```
hits 2 different pods without making it clear that's what happened.